ray-operator/controllers/ray/common/pod.go

```diff
@@ -691,6 +700,13 @@ func setContainerEnvVars(pod *corev1.Pod, rayNodeType rayv1.RayNodeType, gcsOpti
 			Name:  utils.RAY_REDIS_ADDRESS,
 			Value: gcsOptions.RedisAddress,
 		})
+		if gcsOptions.RedisPassword != nil {
+			container.Env = utils.UpsertEnvVar(container.Env, corev1.EnvVar{
+				Name:      utils.REDIS_PASSWORD,
+				Value:     gcsOptions.RedisPassword.Value,
+				ValueFrom: gcsOptions.RedisPassword.ValueFrom,
+			})
+		}
```

Comment:
If `gcsOptions.RedisPassword` is present, upsert an env var `REDIS_PASSWORD`.  It will replace the env var on the value of `--redis-password` flag by shell to pass to redis for authentication.